cmd/stellar-rpc/internal/db/event.go

```diff
@@ -109,80 +116,149 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 			return err
 		}
 
-		if !tx.Result.Successful() {
-			continue
-		}
-
-		transactionHash := tx.Result.TransactionHash[:]
-
+		// Note that we do not skip failed transactions because they still
+		// contain events (e.g., fees are paid regardless of success).
 		allEvents, err := tx.GetTransactionEvents()
 		if err != nil {
 			return err
 		}
 
-		diagEvents := transactionEventsIntoDiagnosticEvents(allEvents)
-
-		if len(diagEvents) == 0 {
-			continue
-		}
-
-		query := sq.Insert(eventTableName).
-			Columns(
-				"id",
-				"contract_id",
-				"event_type",
-				"event_data",
-				"ledger_close_time",
-				"transaction_hash",
-				"topic1", "topic2", "topic3", "topic4",
-			)
-
-		for index, e := range diagEvents {
-			var contractID []byte
-			if e.Event.ContractId != nil {
-				contractID = e.Event.ContractId[:]
-			}
-			index32 := uint32(index) //nolint:gosec
-			id := protocol.Cursor{Ledger: lcm.LedgerSequence(), Tx: tx.Index, Op: 0, Event: index32}.String()
-			eventBlob, err := e.MarshalBinary()
-			if err != nil {
-				return err
+		opEvents := allEvents.OperationEvents
```

Comment:
it might be worthwhile to compose the event loops as functions to keep the insert function a bit thinner and focused on getting the final model into the db:
```
insertableEvents =  gatherTxEvents(insertableEvents, allEvents.TransactionEvents);
insertableEvents =  gatherOpEvents(insertableEvents, allEvents.OperationEvents);
```

Reply:
you're call, I didn't see err's from the loops, but yeah, two more params
```
insertableEvents =  gatherTxEvents(insertableEvents, allEvents.TransactionEvents, tx, sequence);
insertableEvents =  gatherOpEvents(insertableEvents, allEvents.OperationEvents, tx, sequence);
```

Reply:
there's the `default:` error in the tx case :crying_cat_face: